### PR TITLE
Improve card schema

### PR DESCRIPTION
**Pull request by maxammann**
This prepares work on the administration and frontend. The graphql schema was updated.

* Added a enum for qrcodes
* Add signature
* rename hashsecret to pepper to have a more unique name

### CODE REVIEW

File: frontend/lib/identification/base_card_details.dart [hunk 1]
```diff
@@ -2,15 +2,15 @@ enum CardType { standard, gold }
 
 class BaseCardDetails {
   final String fullName;
-  final String hashSecretBase64;
+  final List<int> pepper;
```
Comment (michael-markl) [inline frontend/lib/identification/base_card_details.dart:5]: Shouldn't this be a UInt8Array or so?

File: frontend/lib/verification/scanner/verification_qr_content_parser.dart [hunk 2]
```diff
@@ -18,41 +17,36 @@ class VerificationParseException extends QrCodeParseException {
...
   // TODO (Max): Refactor into Dart extension
-  CardVerifyCode verifyCode;
+  QrCode qrcode;
   try {
-    verifyCode = CardVerifyCode.fromBuffer(rawProtobufData);
-  } on Exception catch (e, stacktrace) {
+    qrcode = QrCode.fromBuffer(base64Decoder.convert(rawBase64Content));
+  } on Exception catch (e, stackTrace) {
     throw VerificationParseException(
       internalMessage: "Failed to parse CardVerifyModel from base64 encoded data. "
```
Comment (michael-markl) [inline frontend/lib/verification/scanner/verification_qr_content_parser.dart:26]:
```suggestion
      internalMessage: "Failed to parse QrCode from base64 encoded data. "
```

File: frontend/lib/verification/scanner/verification_qr_content_parser.dart [hunk 3]
```diff
@@ -18,41 +17,36 @@ class VerificationParseException extends QrCodeParseException {
...
           "Message: ${e.toString()}",
       cause: e,
-      stackTrace: stacktrace,
+      stackTrace: stackTrace,
     );
   }
 
+  if (!qrcode.hasDynamicVerify()) {
```
Comment (michael-markl) [inline frontend/lib/verification/scanner/verification_qr_content_parser.dart:33]:
```suggestion
  // TODO: Allow to parse and verify StaticVerifyCodes.
  if (!qrcode.hasDynamicVerify()) {
```

File: frontend/lib/verification/scanner/verification_encoder.dart [hunk 4]
```diff
@@ -19,26 +18,9 @@ String encodeVerificationCardDetails(VerificationCardDetails verificationCardDet
...
-  final fullName = cardInfo.fullName;
-  final randomBytes = const Base64Encoder().convert(verifyCode.hashSecret);
-  final expirationDay = cardInfo.expirationDay;
-  final cardType = CardType.values[cardInfo.extensions.extensionBavariaCardType.cardType.value];
-  final regionId = cardInfo.extensions.extensionRegion.regionId;
-  final otp = verifyCode.otp;
-
-  return VerificationCardDetails(BaseCardDetails(fullName, randomBytes, expirationDay, cardType, regionId), otp);
+  return const Base64Encoder().convert(QrCode(dynamicVerify: verifyCode).writeToBuffer());
 }
```
Comment (michael-markl) [inline frontend/lib/verification/scanner/verification_encoder.dart:26]: Where did this go? Was this not used anywhere?

**Review by michael-markl — APPROVED** [review]

File: administration/src/components/create-cards/PdfFactory.ts [hunk 5]
```diff
@@ -1,165 +1,167 @@
...
+
+  doc.setFontSize(12)
+  doc.textWithLink(
+    'Öffnen Sie den folgenden Link, um die App herunterzuladen:\nhttps://ehrenamtskarte.app/download_app',
+    width / 2,
+    pageBottom,
+    {
+      url: 'https://ehrenamtskarte.app/download_app',
+      align: 'center',
+    }
```
[the remark this answers is not in the document]
Comment (maxammann) [reply administration/src/components/create-cards/PdfFactory.ts:109]: Done in another PR

File: frontend/lib/verification/scanner/verification_encoder.dart (hunk 4 above)
Comment (maxammann) [reply frontend/lib/verification/scanner/verification_encoder.dart:26]: Nope, dead code

File: frontend/lib/identification/base_card_details.dart (hunk 1 above)
Comment (maxammann) [reply frontend/lib/identification/base_card_details.dart:5]: Nope, it seems like binary data ist List<int> in dart

Comment (michael-markl) [reply frontend/lib/identification/base_card_details.dart:5]: There is Uint8List though which seems more appropriate at first glance.
https://api.dart.dev/be/180791/dart-typed_data/Uint8List-class.html

Comment (michael-markl) [reply frontend/lib/identification/base_card_details.dart:5]: Ah, weirdly, Uint8List implements List<int>. Wild.

**Pull request merged**